static/app/components/organizations/pageFilters/utils.tsx

```diff
@@ -120,10 +124,85 @@ export function isSelectionEqual(selection: PageFilters, other: PageFilters): bo
   return true;
 }
 
+type ProjectId = string | number;
+type EnvironmentId = Environment['id'];
+
+type UpdateData = {
+  project?: ProjectId[] | null;
+  environment?: EnvironmentId[] | null;
+};
+
+/**
+ * Updates the localstorage page filters data
+ *
+ * e.g. if localstorage is empty, user loads issue details for project "foo"
+ * this should not consider "foo" as last used and should not save to local
+ * storage.
+ *
+ * However, if user then changes environment, it should...? Currently it will
+ * save the current project alongside environment to local storage. It's
+ * debatable if this is the desired behavior.
+ *
+ * This will be a no-op if a inaccessible organization slug is passed.
+ */
+export function setPageFiltersStorage(
+  orgSlug: string | null,
+  current: PageFilters,
+  update: UpdateData
+) {
+  const org = orgSlug && OrganizationsStore.get(orgSlug);
+
+  // Do nothing if no org is loaded or user is not an org member. Only
+  // organizations that a user has membership in will be available via the
+  // organizations store
+  if (!org) {
```

Comment:
Should we also ignore superuser actions? Or is this doing that?

Reply:
Good question. This was carried over from the previous implementation in the `PageFiltersStore`